astro-docs/src/content/docs/extending-nx/task-running-lifecycle.mdoc

```diff
@@ -117,10 +117,101 @@ export async function postTasksExecution(options, context) {
 }
 ```
 
+## Using `context.argv` to Determine the Command
+
+The context object includes an `argv` property that contains the original CLI arguments used to invoke Nx. This allows you to distinguish between different execution modes and apply conditional logic.
+
+### Context Properties
+
+Both `preTasksExecution` and `postTasksExecution` hooks receive a context object with the following properties:
+
+```typescript
+// preTasksExecution context
+{
+  id: string;
+  workspaceRoot: string;
+  nxJsonConfiguration: NxJsonConfiguration;
+  argv: string[];  // Original CLI arguments
+}
+
+// postTasksExecution context (includes additional properties)
+{
+  id: string;
+  workspaceRoot: string;
+  nxJsonConfiguration: NxJsonConfiguration;
+  taskResults: TaskResults;
+  argv: string[];  // Original CLI arguments
+  startTime: number;
+  endTime: number;
+}
```

Comment:
maybe define as a 'type' and not a bare object. 

````suggestion
```typescript
type PreTaskExecutionContext = {
  id: string;
  workspaceRoot: string;
  nxJsonConfiguration: NxJsonConfiguration;
  argv: string[];  // Original CLI arguments
}


type PostTaskExecutionContext = {
  id: string;
  workspaceRoot: string;
  nxJsonConfiguration: NxJsonConfiguration;
  taskResults: TaskResults;
  argv: string[];  // Original CLI arguments
  startTime: number;
  endTime: number;
}
````